assemblies/assembly-planner-java.adoc

```diff
@@ -0,0 +1,750 @@
+            for (Lesson lesson : unassignedLessons) {
+                LOGGER.info("  " + lesson.getSubject() + " - " + lesson.getTeacher() + " - " + lesson.getStudentGroup());
+            }
+        }
+    }
+
+}
+----
+
+
+. Run the `TimeTableApp` class as the main class of a normal Java application. The following output should result:
++
+----
+...
+INFO  |            | Room A     | Room B     | Room C     |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 08:30  | English    | Math       |            |
+INFO  |            | I. Jones   | A. Turing  |            |
+INFO  |            | 9th grade  | 10th grade |            |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 09:30  | History    | Physics    |            |
+INFO  |            | I. Jones   | M. Curie   |            |
+INFO  |            | 9th grade  | 10th grade |            |
+...
+----
+
+. Verify the console output. Does it conform to all hard constraints?
+What happens if you comment out the `roomConflict` constraint in `TimeTableConstraintProvider`?
+
+The `info` log shows what OptaPlanner did in those five seconds:
+
+[source,options="nowrap"]
+----
+... Solving started: time spent (33), best score (-8init/0hard/0soft), environment mode (REPRODUCIBLE), random (JDK with seed 0).
+... Construction Heuristic phase (0) ended: time spent (73), best score (0hard/0soft), score calculation speed (459/sec), step total (4).
+... Local Search phase (1) ended: time spent (5000), best score (0hard/0soft), score calculation speed (28949/sec), step total (28398).
+... Solving ended: time spent (5000), best score (0hard/0soft), score calculation speed (28524/sec), phase total (2), environment mode (REPRODUCIBLE).
+----
+
+== Test the application
+A good application includes test coverage. Test the constraints and the solver in your timetable project.
+
+include::{enterprise-dir}/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-test-solver-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-logging-quarkus-proc.adoc[leveloffset=+1]
+
+== Using Micrometer and Prometheus to monitor your school timetable OptaPlanner Java application
+
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with Prometheus to monitor the OptaPlanner solver in the school timetable application.
+
+To configuring a plain Java OptaPlanner application to use Micrometer, you must
+
+
+.Prerequisites
+* You have created the OptaPlanner school timetable application with Java.
+
+.Procedure
+. Add the Micrometer Promtheus dependencies to the school timetable `pom.xml` file:
++
+[source]
+----
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-registry-prometheus</artifactId>
+ <version>1.7.3</version>
```

Comment:
I do not know what version shell we suggest here to our customers 
plain java has 3 options: use transitive dependency, use optaplanner-build-parent as parent, or set the version manually

Reply:
Lets then do not use certain version 1.7.3 here (and for micrometer core) and provide there some version stub.
bellow it could be mentioned in a side note that micrometer-core could be used transitively from optaplanner-core or 
micrometer-registry-prometheus and  micrometer-core could be  inherited from optaplanner-build-parent

Reply:
I need a bit more information about this:
 micrometer-core could be used transitively from optaplanner-core 

Can you explain it please?
What is optaplanner-build-parent?

Reply:
optaplanner-build-parent is a pom file with declared versions which is used during certification

"micrometer-core could be used transitively from optaplanner-core" means that optaplanner-core already has this dependency and user do not need to define it in own pom file, but rather use it as a part of optaplanner-core

Reply:
Updated

Reply:
LGTM